catalog/core/catalog-core-solrcommands/src/main/java/org/codice/ddf/commands/solr/RestoreCommand.java

```diff
@@ -0,0 +1,318 @@
+      uriBuilder.addParameter("command", "restore");
+
+      if (StringUtils.isNotBlank(backupLocation)) {
+        uriBuilder.addParameter("location", backupLocation);
+      }
+
+      URI restoreUri = uriBuilder.build();
+      LOGGER.debug("Sending request to {}", restoreUri);
+
+      printResponse(sendGetRequest(restoreUri));
+    } catch (IOException | SolrServerException e) {
+      LOGGER.info("Unable to perform single node Solr restore, core: {}", coreName, e);
+    }
+  }
+
+  private void performSolrCloudRestore() throws IOException {
+    SolrClient client = null;
+    try {
+      client = getCloudSolrClient();
+      if (status) {
+        verifyStatusInput();
+        printStatus(client, requestId);
+      } else {
+        verifyRestoreInput();
+        performSolrCloudRestore(client);
+      }
+    } finally {
+      shutdown(client);
+    }
+  }
+
+  private void printResponse(@Nullable HttpResponse response) {
+    if (response != null) {
+      StatusLine statusLine = response.getStatusLine();
+      if (statusLine.getStatusCode() == HttpStatus.SC_OK) {
+        printSuccessMessage(String.format("%nRestore of [%s] complete.%n", coreName));
+      } else {
+        printErrorMessage(String.format("Error restoring Solr core: [%s]", coreName));
+        printErrorMessage(
+            String.format(
+                "Restore command failed due to: %d - %s",
+                statusLine.getStatusCode(), statusLine.getReasonPhrase()));
+        printErrorMessage(String.format("Request: %s", getReplicationUrl(coreName)));
+      }
+    }
+  }
+
+  private boolean restore(
+      SolrClient client, String collection, String backupLocation, String backupName)
+      throws IOException, SolrServerException {
+    if (!canRestore(client, collection)) {
+      LOGGER.warn("Unable to restore collection {}", collection);
+      return false;
+    }
+
+    CollectionAdminRequest.Restore restore =
+        CollectionAdminRequest.AsyncCollectionAdminRequest.restoreCollection(collection, backupName)
+            .setLocation(backupLocation);
+
+    String syncReqId = restore.processAsync(client);
+
+    while (true) {
+      CollectionAdminRequest.RequestStatusResponse requestStatusResponse =
+          CollectionAdminRequest.requestStatus(syncReqId).process(client);
+      RequestStatusState requestStatus = requestStatusResponse.getRequestStatus();
+      if (requestStatus == RequestStatusState.COMPLETED) {
+        LOGGER.debug("Restore status: {}", requestStatus);
+        return true;
+      } else if (requestStatus == RequestStatusState.FAILED
+          || requestStatus == RequestStatusState.NOT_FOUND) {
+        LOGGER.debug("Restore status: {}", requestStatus);
+        printErrorMessage("Restore failed. ");
+        printResponseErrorMessages(requestStatusResponse);
+        return false;
+      }
+      try {
+        TimeUnit.SECONDS.sleep(1);
+      } catch (InterruptedException e) {
+        Thread.currentThread().interrupt();
```

Comment:
❗️ You do not want to re-interrupt the thread within the loop as this will simply trigger the same interruption yet again when it loops back. You need to set a variable to true to remember you were interrupted and then put the `while {}` inside a `try/finally` and in the `finally` check if you variable was set and if so then re-interrupt the current thread as you do to propagate the interruption.